src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8622,7 +8620,20 @@ internal static List<AlternateStreamData> GetStreams(string path)
             SafeFindHandle handle = NativeMethods.FindFirstStreamW(
                 path, NativeMethods.StreamInfoLevels.FindStreamInfoStandard,
                 findStreamData, 0);
-            if (handle.IsInvalid) throw new Win32Exception();
+            if (handle.IsInvalid)
+            {
+                // Except for a test for handle.IsInvalid, this must be immediately after a P/Invoke call.
+                int error = Marshal.GetLastWin32Error();
+                // Directories don't normally have alternate streams, so this is not an exceptional state.
```

Comment:
Fix CodeFactor issue:
```suggestion
                int error = Marshal.GetLastWin32Error();

                // Directories don't normally have alternate streams, so this is not an exceptional state.
```

Reply:
Fixed in next push.